finesse/gui/serial_view.py

```diff
@@ -1,138 +1,206 @@
-                devices to control and as values a dictionary with the port and baud
-                rate chosen for that device.
-            avail_ports: Sequence of possible serial ports.
-            avail_baud_rates: Sequence of possible baud rates.
-        """
-        super().__init__("Serial port control")
-
-        self._devices = deepcopy(devices)
-        self._is_open = {d: False for d in devices.keys()}
-        layout = self._create_controls(avail_ports, avail_baud_rates)
-        self.setLayout(layout)
+        The controls are added as a new row to a QGridLayout.
 
-    def _create_controls(
-        self, avail_ports: Sequence[str], avail_baud_rates: Sequence[int]
-    ) -> QGridLayout:
-        """Creates the controls for the ports of the devices.
+        The initially selected port and baudrate are loaded from the program settings if
+        available, otherwise the first available port and the device's default baudrate
+        are used, respectively.
 
         Args:
-            avail_ports: Sequence of possible serial ports.
-            avail_baud_rates: Sequence of possible baud rates.
-
-        Returns:
-            QGridLayout: The layout with the widgets.
+            layout: The QGridLayout to add the controls to
+            row: The row of the QGridLayout to add the controls to
+            device: The device these controls refer to
+            avail_ports: Possible serial ports to choose from
+            avail_baudrates: Possible baudrates to choose from
         """
-        layout = QGridLayout()
-        for i, (name, details) in enumerate(self._devices.items()):
-            # Adding the device name
-            layout.addWidget(QLabel(name), i, 0)
-
-            # Its port
-            _port = QComboBox()
-            _port.addItems(list(avail_ports))
-            _port.setCurrentText(details["port"])
-            _port.currentTextChanged.connect(  # type: ignore
-                partial(self.on_port_changed, device_name=name)
-            )
-            layout.addWidget(_port, i, 1)
-
-            # Its baud rate
-            _brate = QComboBox()
-            _brate.addItems([str(br) for br in avail_baud_rates])
-            _brate.setCurrentText(details["baud_rate"])
-            _brate.currentTextChanged.connect(  # type: ignore
-                partial(self.on_baud_rate_changed, device_name=name)
-            )
-            layout.addWidget(_brate, i, 2)
-
-            # And a manual way to open/close it
-            _open_close_btn = QPushButton("Open")
-            _open_close_btn.setCheckable(True)
-            _open_close_btn.released.connect(  # type: ignore
-                partial(
-                    self.on_open_close_clicked, device_name=name, button=_open_close_btn
-                )
-            )
-            _open_close_btn.setChecked(False)
-            layout.addWidget(_open_close_btn, i, 3)
+        self.name = device.name
+        self.label = device.label
```

Comment:
Why not storing this as `self.device = device`? It seems a waste to define a data class to put together the information concerning a device and then unpack it almost immediately.